R/Achilles.R

```diff
@@ -1006,6 +1029,73 @@ dropAllScratchTables <- function(connectionDetails,
   ParallelLogger::unregisterLogger("dropAllScratchTables")
 }
 
+#' Optimize atlas cache
+#'
+#' @details
+#' Post-processing, optimize data for atlas cache in separate table to help performance.
+#'
+#' @param connectionDetails                An R object of type \code{connectionDetails} created using the function \code{createConnectionDetails} in the \code{DatabaseConnector} package.
+#' @param resultsDatabaseSchema		       Fully qualified name of database schema that we can write final results to. Default is cdmDatabaseSchema.
+#'                                         On SQL Server, this should specifiy both the database and the schema, so for example, on SQL Server, 'cdm_results.dbo'.
+#' @param vocabDatabaseSchema
+#' @param outputFolder                     Path to store logs and SQL files
+#' @param sqlOnly                          TRUE = just generate SQL files, don't actually run, FALSE = run Achilles
+#' @param verboseMode                      Boolean to determine if the console will show all execution steps. Default = TRUE
+#' @param tempAchillesPrefix               The prefix to use for the "temporary" (but actually permanent) Achilles analyses tables. Default is "tmpach"
+#'
+#' @export
+optimizeAtlasCache <- function(connectionDetails,
+                               resultsDatabaseSchema,
+                               vocabDatabaseSchema = resultsDatabaseSchema,
+                               outputFolder = "output",
+                               sqlOnly = FALSE,
+                               verboseMode = TRUE,
+                               tempAchillesPrefix = "tmpach") {
+
+  # Log execution --------------------------------------------------------------------------------------------------------------------
+
```

Comment:
add logic to create the outputFolder if it doesn't exist

Reply:
fixed

Reply:
I'm not sure this works. When I tested on my machine, naming an outputFolder that didn't exist, the logging failed. I'm thinking instead using this at the top of the function:

```
  if (!dir.exists(outputFolder)) {
    dir.create(path = outputFolder, recursive = TRUE)
  }
```

Reply:
I don't understand
I tried this on my machine and it worked
```optimizeAtlasCache(connectionDetails,
         resultsDatabaseSchema="results",
         vocabDatabaseSchema = "dbo",
         sqlOnly=FALSE,
         outputFolder="test_output")   ```